app/assets/javascripts/discourse/lib/utilities.js.es6

```diff
@@ -410,5 +410,46 @@ export function rescueThemeError(name, error, api) {
   document.body.prepend(alertDiv);
 }
 
+const CODE_BLOCKS_RULES = [
+  { rule: "^(    |\\t).*", end: "" },
+  { rule: "`.+?`", end: "`" },
+  { rule: "^```[^]*?^```", end: "\n```" },
+  { rule: "\\[code\\][^]*?\\[\\/code\\]", end: "\n[/code]" }
+];
+
+export function inCodeBlock(text, pos) {
+  const regexp = new RegExp(
+    CODE_BLOCKS_RULES.map(rules => rules.rule).join("|"),
+    "gm"
+  );
+
+  let match;
+  while ((match = regexp.exec(text)) != null) {
+    const begin = match.index;
+    const end = match.index + match[0].length;
+    if (begin <= pos && pos <= end) {
+      return true;
+    }
+  }
+
+  // Try to end block and see if other code blocks are found
```

Comment:
hmmm do we want to do this? I get that having auto complete after "test /` @" can be arguably allowed but honestly I am fine to just ban it in this case.

Reply:
The user can still end the code block before proceed with its contents, so I will just get rid of this.